pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+    independent variable) via ``time_col``. The fastest operation
+    occurs when the input time series are equally spaced and have no gaps or
+    outliers and the special options are not needed.
+    Read a table and output as a :class:`numpy.ndarray`,
+    :class:`pandas.DataFrame`, or ASCII file.
+
+    Full option list at :gmt-docs:`filter1d.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    filter : str
+        **type**\ *width*\ [**+h**].
+        Sets the filter **type**. Choose among convolution and non-convolution
+        filters. Append the filter code followed by the full filter
+        *width* in same units as time column. By default, this
+        performs a low-pass filtering; append **+h** to select high-pass
+        filtering. Some filters allow for optional arguments and a modifier.
+
+        Available convolution filter types are:
+
+        - (**b**) Boxcar: All weights are equal.
+        - (**c**) Cosine Arch: Weights follow a cosine arch curve.
+        - (**g**) Gaussian: Weights are given by the Gaussian function.
+        - (**f**) Custom: Instead of *width* give name of a one-column file
+          with your own weight coefficients.
+
+        Non-convolution filter types are:
+
+        - (**m**) Median: Returns median value.
+        - (**p**) Maximum likelihood probability (a mode estimator): Return
+          modal value. If more than one mode is found we return their average
+          value. Append **+l** or **+u** if you rather want
+          to return the lowermost or uppermost of the modal values.
+        - (**l**) Lower: Return the minimum of all values.
+        - (**L**) Lower: Return minimum of all positive values only.
+        - (**u**) Upper: Return maximum of all values.
+        - (**U**) Upper: Return maximum of all negative values only.
+
+        Upper case type **B**, **C**, **G**, **M**, **P**, **F** will use
+        robust filter versions: i.e., replace outliers (2.5 L1 scale off
+        median, using 1.4826 \* median absolute deviation [MAD]) with median
+        during filtering.
+
+        In the case of **L**\|\ **U** it is possible that no data passes
+        the initial sign test; in that case the filter will return 0.0.
+        Apart from custom coefficients (**f**), the other filters may accept
+        variable filter widths by passing *width* as a two-column time-series
+        file with filter widths in the second column.  The filter-width file
+        does not need to be co-registered with the data as we obtain the
+        required filter width at each output location via interpolation.  For
+        multi-segment data files the filter file must either have the same
+        number of segments or just a single segment to be used for all data
+        segments.
+
+    end : bool
+        Include Ends of time series in output. The default [False] loses
+        half the filter-width of data at each end.
+
+    time_col : int or str
```

Comment:
Is a `str` allowed for time_col (`-N`)? As in users can pass in the name of the time column in the `pandas.DataFrame` table input?

Reply:
I am under the impression that GMT reads a string containing an integer or floating point number as the same value as if it was passed as a number instead. I don't know why a Python user would ever pass a number as a string, so it may just be easier to leave it as `int`.

I'll see if I can try and test it, but `filter1d` is one (of the many) module that I don't understand its application and just wrapped using the provided GMT example and dataset.

Reply:
> As in users can pass in the name of the time column in the `pandas.DataFrame` table input?

I think this is a nice feature and can be implemented in a separate PR. In this initial wrapper, a time_col as `int` should be good.